server/src/processor.rs

```diff
@@ -213,17 +214,22 @@ impl <'a> SignalProcessor<'a> {
     }
 
     fn generate_response(self) -> ResponseEntity {
-        let stats = self.stats();
+        let mut stats_option = None;
+        let mut stats = self.stats();
         let settled = self.current_settled
             .into_iter()
             .map(|(_, v)| v)
             .collect::<Vec<Signal>>();
 
+        if stats.yes != 0. && stats.no != 0. {
+            stats_option = Some(stats);
+        }
+
         ResponseEntity {
             settled,
             unsettled: self.unsettled_signals,
             invalid: self.invalid_signals,
-            stats
+            stats: stats_option
         }
     }
```

Comment:
You can match directly on the `ResponseEntity` struct without having to shadow `None` with `stats_option`.

`fn generate_response(self) -> ResponseEntity {
        let stats = self.stats();
        let settled = self
            .current_settled
            .into_iter()
            .map(|(_, v)| v)
            .collect::<Vec<Signal>>();

        ResponseEntity {
            settled,
            unsettled: self.unsettled_signals,
            invalid: self.invalid_signals,
            stats: match stats.yes != 0. && stats.no != 0. {
                true => Some(stats),
                false => None,
            },
        }
    }`

Reply:
And another change would be required on `models/mod.rs - line 73`:
`self.stats = stats` => ` self.stats = Some(stats)`